File: tests.php5/arrayToString.php
#!/usr/bin/php

<?php
include_once ("java/Java.inc");

$here=realpath(dirname($_SERVER["SCRIPT_FILENAME"]));
if(!$here) $here=getcwd();
$ArrayToString = new java_class("ArrayToString");


// create long array ...
$length=10;
for($i=0; $i<$length; $i++) {
  $arr[$i]=$i;
}

// ... and post it to java.  Should print integers
print "integer array: ". java_values($ArrayToString->arrayToString($arr)) . "<br>\n";


// double
for($i=0; $i<$length; $i++) {
  $arr[$i]=$i +1.23;
}

// should print doubles
print "double array: ". java_values($ArrayToString->arrayToString($arr)) . "<br>\n";


// boolean
for($i=0; $i<$length; $i++) {
  $arr[$i]=$i%2?true:false;
}

// should print booleans
print "boolean array: ". java_values($ArrayToString->arrayToString($arr)) ."<br>\n";

?>

File: legacy/tests.php5/ArrayToString.java
public class ArrayToString {
    public static String arrayToString(String[] arr) {
	StringBuffer b = new StringBuffer(arr.length);
	for(int i=0; i<arr.length; i++) {
	    b.append(String.valueOf(arr[i]));
	    if(i+1<arr.length) b.append(" ");
	}
	return b.toString();
    }
    public static String arrayToString(int[] arr) {
	StringBuffer b = new StringBuffer(arr.length);
	for(int i=0; i<arr.length; i++) {
	    b.append(String.valueOf(arr[i]));
	    if(i+1<arr.length) b.append(" ");
	}
	return b.toString();
    }
    public static String arrayToString(double[] arr) {
	StringBuffer b = new StringBuffer(arr.length);
	for(int i=0; i<arr.length; i++) {
	    b.append(String.valueOf(arr[i]));
	    if(i+1<arr.length) b.append(" ");
	}
	return b.toString();
    }
    public static String arrayToString(boolean[] arr) {
	StringBuffer b = new StringBuffer(arr.length);
	for(int i=0; i<arr.length; i++) {
	    b.append(String.valueOf(arr[i]));
	    if(i+1<arr.length) b.append(" ");
	}
	return b.toString();
    }
}
